README.md

```diff
@@ -26,7 +26,7 @@ or a map of the shape:
 
 ```erlang
 #{
-  provider_source => Provider :: atom(),
+  provider_source => Provider :: module(),
```

Comment:
The `module()` type is effectively the same as the `atom()` type. However, I think this change clarifies that it represents a module name to the user.